src/libraries/System.Text.Json/tests/System.Text.Json.Tests/NewtonsoftTests/ImmutableCollectionsTests.cs

```diff
@@ -316,7 +316,7 @@ public void DeserializeSortedSet()
   ""3""
 ]";
 
-            ImmutableSortedSet<string> data = JsonSerializer.Deserialize<ImmutableSortedSet<string>>(json);
+            ISet<string> data = JsonSerializer.Deserialize<ImmutableSortedSet<string>>(json);
```

Comment:
Why did these change?

Reply:
Ah, interesting case!

So, this is a breaking change, but not one caused by me. The latest source for Xunit.Assert contains an overload for both `IReadOnlySet` and `ISet`. That makes this overload ambiguous.

I'm not sure the best way to handle this. This was the easiest way I could think to make progress.

For options I think we could:

1. Make the change above
2. Revert the xunit source to an older version which doesn't have this change
3. Raise the change as an issue in the xunit.assert repo as a potential breaking change.

Do you have any preferences?

Reply:
Seems reasonable to just accommodate the change.